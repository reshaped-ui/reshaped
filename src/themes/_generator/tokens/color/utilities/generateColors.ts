import type { ThemeDefinition } from "themes/_generator/tokens/types";
import {
	Token,
	HexColor,
	OklchToken,
	type Name as TokenName,
	type ColorValue,
	Hue,
	OklchColor,
} from "../color.types";
import { hexToOklch, tokenToOklchToken } from "./convert";

const parseColor = (color: HexColor | Token) => {
	const isString = typeof color === "string";

	if (!isString && "oklch" in color && color.oklch) {
		return {
			light: color.oklch,
			dark: color.oklchDark,
		};
	}

	const hexLight = isString ? color : color.hex;
	const hexDark = isString ? undefined : color.hexDark;

	const light = hexToOklch(hexLight);
	const dark = hexDark ? hexToOklch(hexDark) : undefined;

	if (!light) {
		throw new Error(`[Reshaped] Failed when parsing color: ${JSON.stringify(color)}`);
	}

	return { light, dark };
};

const getDarkModeColor = (lch: OklchColor) => {
	const { l, c } = lch;

	// Top boundary at which background colors are perceived as vibrant
	const vibrancyThreshold = 0.25;
	// A linear modifier of how gray a color is, gray = 1, vibrant = 0
	const neutralFactor = Math.min(1, Math.max(0, (vibrancyThreshold - c) / vibrancyThreshold));
	// Non-linear modification to make gray color inversion progress faster
	const easedVibrancy = 1 - Math.pow(neutralFactor, 2);

	const invertedL = 1 - l;
	// Take plain inversion and increase it back by vibrancy modifier
	// Vibrant colors should get back to almost the original value
	// while non-briant colors should lean towards full inversion

	// Example:
	// Black should become white in dark mode
	// Light gray should become dark gray
	// Vibrant blue should stay vibrant blue
	const darkL = invertedL + (l - invertedL) * easedVibrancy;

	// Make sure dark mode value doesn't become too dark and is still visible on dark page background
	const clampedDarkL = Math.max(0.36, darkL);

	return { ...lch, c: c * 0.9, l: clampedDarkL };
};

const generateColorValues = (
	key: string,
	token: ColorValue
): Partial<Record<TokenName, OklchToken>> => {
	const { light: bg, dark } = parseColor(token);
	const capitalizedKey = key.charAt(0).toUpperCase() + key.slice(1);
	const neutral = key === "neutral";
	const warning = key === "warning";

	const bgDark = dark || getDarkModeColor(bg);
	const bgFaded = {
		...bg,
		l: 0.97,
		c: neutral ? 0.005 : warning ? 0.04 : 0.02,
	};
	const bgFadedDark = {
		...bgDark,
		l: 0.25,
		// For primary color with low chroma, we still have to make sure it stays low
		c: neutral ? 0.01 : bgDark.c / 4,
	};

	const fg = neutral ? { ...bg, l: 0.2 } : { ...bg, l: 0.5 };
	const fgDark = neutral ? { ...bgDark, l: 0.96 } : { ...bgDark, l: 0.75, c: bg.c * 0.85 };

	const bd = neutral ? { ...bg, l: 0, alpha: 0.12 } : { ...bg, l: bg.l - 0.08 };
	const bdDark = neutral ? { ...bgDark, l: 1, alpha: 0.16 } : { ...bgDark, l: bgDark.l + 0.08 };
	const bdFaded = neutral
		? { ...bgFaded, l: 0, alpha: 0.08 }
		: { ...bgFaded, l: bgFaded.l - 0.05, c: bgFaded.c + 0.01 };
	const bdFadedDark = neutral
		? { ...bgFadedDark, l: 1, alpha: 0.08 }
		: { ...bgFadedDark, l: bgFadedDark.l + 0.1 };

	const output = {
		[`background${capitalizedKey}`]: {
			oklch: bg,
			oklchDark: bgDark,
		},
		[`background${capitalizedKey}Faded`]: {
			oklch: bgFaded,
			oklchDark: bgFadedDark,
		},
		[`border${capitalizedKey}`]: {
			oklch: bd,
			oklchDark: bdDark,
		},
		[`border${capitalizedKey}Faded`]: {
			oklch: bdFaded,
			oklchDark: bdFadedDark,
		},
		[`foreground${capitalizedKey}`]: {
			oklch: fg,
			oklchDark: fgDark,
		},
	};

	if (neutral) {
		output[`foreground${capitalizedKey}Faded`] = {
			oklch: { ...fg, l: fg.l + 0.25 },
			oklchDark: { ...fgDark, l: fgDark.l - 0.15 },
		};
		output[`backgroundDisabled`] = {
			oklch: { ...bg, l: 0.95, c: 0 },
			oklchDark: { ...bgDark, l: 0.28, c: 0 },
		};
		output[`backgroundDisabledFaded`] = {
			oklch: { ...bg, l: 0.98, c: 0 },
			oklchDark: { ...bgDark, l: 0.23, c: 0 },
		};
		output[`borderDisabled`] = {
			oklch: { ...bd, alpha: 0.06 },
			oklchDark: { ...bgDark, l: 0.28, c: 0 },
		};
		output[`foregroundDisabled`] = {
			oklch: { ...fg, l: 0.84, c: 0 },
			oklchDark: { ...fgDark, l: 0.4, c: 0 },
		};
		output[`backgroundElevationBase`] = {
			oklch: { ...bg, l: 1, c: 0 },
			oklchDark: { ...bgDark, l: 0.2, c: 0 },
		};
		output[`backgroundElevationRaised`] = {
			oklch: { ...bg, l: 1, c: 0 },
			oklchDark: { ...bgDark, l: 0.21 },
		};
		output[`backgroundElevationOverlay`] = {
			oklch: { ...bg, l: 1, c: 0 },
			oklchDark: { ...bgDark, l: 0.22, c: 0 },
		};
		output[`backgroundPage`] = {
			oklch: { ...bg, l: 1, c: 0 },
			oklchDark: { ...bgDark, l: 0.16, c: 0 },
		};
		output[`backgroundPageFaded`] = {
			oklch: { ...bg, l: 0.97, c: 0 },
			oklchDark: { ...bgDark, l: 0.18, c: 0 },
		};
	}

	return output;
};

const getOklchToken = (color: ColorValue) => {
	const token = typeof color === "string" ? { hex: color } : color;
	return tokenToOklchToken(token);
};

const generateColors = (args: Partial<Record<Hue, ColorValue>> = {}) => {
	const {
		primary = "#1469ff",
		critical = "#e22c2c",
		warning = "#facc15",
		positive = "#118850",
		neutral = "#e3e4e8",
		brand,
	} = args;

	return {
		...generateColorValues("primary", getOklchToken(primary)),
		...generateColorValues("critical", getOklchToken(critical)),
		...generateColorValues("warning", getOklchToken(warning)),
		...generateColorValues("positive", getOklchToken(positive)),
		...generateColorValues("neutral", getOklchToken(neutral)),
		brand: getOklchToken(brand || primary),
		white: getOklchToken("#ffffff"),
		black: getOklchToken("#000000"),
	} as ThemeDefinition["color"];
};

export default generateColors;
