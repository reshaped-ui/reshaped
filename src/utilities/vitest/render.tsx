import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
