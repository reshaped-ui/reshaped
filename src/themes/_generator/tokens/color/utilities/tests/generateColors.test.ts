import { expect, test, describe } from "vitest";
import generateColors from "../generateColors";

describe("themes/tokens/color/utilities/generateColors", () => {
	test("with default values", () => {
		expect(generateColors()).toMatchSnapshot();
	});

	test("with a passed value", () => {
		const black = { l: 0, c: 0, h: 0 };
		const colors = generateColors({ primary: { oklch: black } });

		expect(colors).toMatchSnapshot();
		expect(colors.backgroundPrimary).toEqual({
			oklch: { ...black, mode: "oklch" },
			oklchDark: { ...black, l: 1, mode: "oklch" },
		});
	});
});
