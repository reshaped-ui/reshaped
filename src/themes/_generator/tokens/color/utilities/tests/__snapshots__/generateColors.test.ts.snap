// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`themes/tokens/color/utilities/generateColors > with a passed value 1`] = `
{
  "backgroundCritical": {
    "oklch": {
      "c": 0.22,
      "h": 26.97,
      "l": 0.59,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.17600000000000002,
      "h": 26.97,
      "l": 0.5874079999999999,
      "mode": "oklch",
    },
  },
  "backgroundCriticalFaded": {
    "oklch": {
      "c": 0.02,
      "h": 26.97,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.0352,
      "h": 26.97,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "backgroundDisabled": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 0.95,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.28,
      "mode": "oklch",
    },
  },
  "backgroundDisabledFaded": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 0.98,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.23,
      "mode": "oklch",
    },
  },
  "backgroundElevationBase": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.2,
      "mode": "oklch",
    },
  },
  "backgroundElevationOverlay": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.22,
      "mode": "oklch",
    },
  },
  "backgroundElevationRaised": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.21,
      "mode": "oklch",
    },
  },
  "backgroundNeutral": {
    "oklch": {
      "c": 0.01,
      "h": 262.67,
      "l": 0.92,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.008,
      "h": 262.67,
      "l": 0.36,
      "mode": "oklch",
    },
  },
  "backgroundNeutralFaded": {
    "oklch": {
      "c": 0.005,
      "h": 262.67,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.01,
      "h": 262.67,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "backgroundPage": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.16,
      "mode": "oklch",
    },
  },
  "backgroundPageFaded": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.18,
      "mode": "oklch",
    },
  },
  "backgroundPositive": {
    "oklch": {
      "c": 0.13,
      "h": 153.78,
      "l": 0.53,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.10400000000000001,
      "h": 153.78,
      "l": 0.516176,
      "mode": "oklch",
    },
  },
  "backgroundPositiveFaded": {
    "oklch": {
      "c": 0.02,
      "h": 153.78,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.020800000000000003,
      "h": 153.78,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "backgroundPrimary": {
    "oklch": {
      "c": 0,
      "h": 0,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 0,
      "l": 1,
      "mode": "oklch",
    },
  },
  "backgroundPrimaryFaded": {
    "oklch": {
      "c": 0.02,
      "h": 0,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 0,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "backgroundWarning": {
    "oklch": {
      "c": 0.2,
      "h": 80,
      "l": 0.83,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.16000000000000003,
      "h": 80,
      "l": 0.8036,
      "mode": "oklch",
    },
  },
  "backgroundWarningFaded": {
    "oklch": {
      "c": 0.04,
      "h": 80,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.03200000000000001,
      "h": 80,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "black": {
    "oklch": {
      "c": 0,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": undefined,
  },
  "borderCritical": {
    "oklch": {
      "c": 0.22,
      "h": 26.97,
      "l": 0.51,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.17600000000000002,
      "h": 26.97,
      "l": 0.6674079999999999,
      "mode": "oklch",
    },
  },
  "borderCriticalFaded": {
    "oklch": {
      "c": 0.03,
      "h": 26.97,
      "l": 0.9199999999999999,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.0352,
      "h": 26.97,
      "l": 0.35,
      "mode": "oklch",
    },
  },
  "borderDisabled": {
    "oklch": {
      "alpha": 0.06,
      "c": 0.01,
      "h": 262.67,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.28,
      "mode": "oklch",
    },
  },
  "borderNeutral": {
    "oklch": {
      "alpha": 0.12,
      "c": 0.01,
      "h": 262.67,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": {
      "alpha": 0.16,
      "c": 0.008,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
  },
  "borderNeutralFaded": {
    "oklch": {
      "alpha": 0.08,
      "c": 0.005,
      "h": 262.67,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": {
      "alpha": 0.08,
      "c": 0.01,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
  },
  "borderPositive": {
    "oklch": {
      "c": 0.13,
      "h": 153.78,
      "l": 0.45,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.10400000000000001,
      "h": 153.78,
      "l": 0.5961759999999999,
      "mode": "oklch",
    },
  },
  "borderPositiveFaded": {
    "oklch": {
      "c": 0.03,
      "h": 153.78,
      "l": 0.9199999999999999,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.020800000000000003,
      "h": 153.78,
      "l": 0.35,
      "mode": "oklch",
    },
  },
  "borderPrimary": {
    "oklch": {
      "c": 0,
      "h": 0,
      "l": -0.08,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 0,
      "l": 1.08,
      "mode": "oklch",
    },
  },
  "borderPrimaryFaded": {
    "oklch": {
      "c": 0.03,
      "h": 0,
      "l": 0.9199999999999999,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 0,
      "l": 0.35,
      "mode": "oklch",
    },
  },
  "borderWarning": {
    "oklch": {
      "c": 0.2,
      "h": 80,
      "l": 0.75,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.16000000000000003,
      "h": 80,
      "l": 0.8835999999999999,
      "mode": "oklch",
    },
  },
  "borderWarningFaded": {
    "oklch": {
      "c": 0.05,
      "h": 80,
      "l": 0.9199999999999999,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.03200000000000001,
      "h": 80,
      "l": 0.35,
      "mode": "oklch",
    },
  },
  "brand": {
    "oklch": {
      "c": 0,
      "h": 0,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": undefined,
  },
  "foregroundCritical": {
    "oklch": {
      "c": 0.22,
      "h": 26.97,
      "l": 0.5,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.187,
      "h": 26.97,
      "l": 0.75,
      "mode": "oklch",
    },
  },
  "foregroundDisabled": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 0.84,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.4,
      "mode": "oklch",
    },
  },
  "foregroundNeutral": {
    "oklch": {
      "c": 0.01,
      "h": 262.67,
      "l": 0.2,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.008,
      "h": 262.67,
      "l": 0.96,
      "mode": "oklch",
    },
  },
  "foregroundNeutralFaded": {
    "oklch": {
      "c": 0.01,
      "h": 262.67,
      "l": 0.45,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.008,
      "h": 262.67,
      "l": 0.8099999999999999,
      "mode": "oklch",
    },
  },
  "foregroundPositive": {
    "oklch": {
      "c": 0.13,
      "h": 153.78,
      "l": 0.5,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.1105,
      "h": 153.78,
      "l": 0.75,
      "mode": "oklch",
    },
  },
  "foregroundPrimary": {
    "oklch": {
      "c": 0,
      "h": 0,
      "l": 0.5,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 0,
      "l": 0.75,
      "mode": "oklch",
    },
  },
  "foregroundWarning": {
    "oklch": {
      "c": 0.2,
      "h": 80,
      "l": 0.5,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.17,
      "h": 80,
      "l": 0.75,
      "mode": "oklch",
    },
  },
  "white": {
    "oklch": {
      "c": 0,
      "l": 0.999999993473546,
      "mode": "oklch",
    },
    "oklchDark": undefined,
  },
}
`;

exports[`themes/tokens/color/utilities/generateColors > with default values 1`] = `
{
  "backgroundCritical": {
    "oklch": {
      "c": 0.22,
      "h": 26.97,
      "l": 0.59,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.17600000000000002,
      "h": 26.97,
      "l": 0.5874079999999999,
      "mode": "oklch",
    },
  },
  "backgroundCriticalFaded": {
    "oklch": {
      "c": 0.02,
      "h": 26.97,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.0352,
      "h": 26.97,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "backgroundDisabled": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 0.95,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.28,
      "mode": "oklch",
    },
  },
  "backgroundDisabledFaded": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 0.98,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.23,
      "mode": "oklch",
    },
  },
  "backgroundElevationBase": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.2,
      "mode": "oklch",
    },
  },
  "backgroundElevationOverlay": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.22,
      "mode": "oklch",
    },
  },
  "backgroundElevationRaised": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.21,
      "mode": "oklch",
    },
  },
  "backgroundNeutral": {
    "oklch": {
      "c": 0.01,
      "h": 262.67,
      "l": 0.92,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.008,
      "h": 262.67,
      "l": 0.36,
      "mode": "oklch",
    },
  },
  "backgroundNeutralFaded": {
    "oklch": {
      "c": 0.005,
      "h": 262.67,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.01,
      "h": 262.67,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "backgroundPage": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.16,
      "mode": "oklch",
    },
  },
  "backgroundPageFaded": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.18,
      "mode": "oklch",
    },
  },
  "backgroundPositive": {
    "oklch": {
      "c": 0.13,
      "h": 153.78,
      "l": 0.53,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.10400000000000001,
      "h": 153.78,
      "l": 0.516176,
      "mode": "oklch",
    },
  },
  "backgroundPositiveFaded": {
    "oklch": {
      "c": 0.02,
      "h": 153.78,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.020800000000000003,
      "h": 153.78,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "backgroundPrimary": {
    "oklch": {
      "c": 0.24,
      "h": 262.67,
      "l": 0.55,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.192,
      "h": 262.67,
      "l": 0.54984,
      "mode": "oklch",
    },
  },
  "backgroundPrimaryFaded": {
    "oklch": {
      "c": 0.02,
      "h": 262.67,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.038400000000000004,
      "h": 262.67,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "backgroundWarning": {
    "oklch": {
      "c": 0.2,
      "h": 80,
      "l": 0.83,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.16000000000000003,
      "h": 80,
      "l": 0.8036,
      "mode": "oklch",
    },
  },
  "backgroundWarningFaded": {
    "oklch": {
      "c": 0.04,
      "h": 80,
      "l": 0.97,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.03200000000000001,
      "h": 80,
      "l": 0.25,
      "mode": "oklch",
    },
  },
  "black": {
    "oklch": {
      "c": 0,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": undefined,
  },
  "borderCritical": {
    "oklch": {
      "c": 0.22,
      "h": 26.97,
      "l": 0.51,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.17600000000000002,
      "h": 26.97,
      "l": 0.6674079999999999,
      "mode": "oklch",
    },
  },
  "borderCriticalFaded": {
    "oklch": {
      "c": 0.03,
      "h": 26.97,
      "l": 0.9199999999999999,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.0352,
      "h": 26.97,
      "l": 0.35,
      "mode": "oklch",
    },
  },
  "borderDisabled": {
    "oklch": {
      "alpha": 0.06,
      "c": 0.01,
      "h": 262.67,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.28,
      "mode": "oklch",
    },
  },
  "borderNeutral": {
    "oklch": {
      "alpha": 0.12,
      "c": 0.01,
      "h": 262.67,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": {
      "alpha": 0.16,
      "c": 0.008,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
  },
  "borderNeutralFaded": {
    "oklch": {
      "alpha": 0.08,
      "c": 0.005,
      "h": 262.67,
      "l": 0,
      "mode": "oklch",
    },
    "oklchDark": {
      "alpha": 0.08,
      "c": 0.01,
      "h": 262.67,
      "l": 1,
      "mode": "oklch",
    },
  },
  "borderPositive": {
    "oklch": {
      "c": 0.13,
      "h": 153.78,
      "l": 0.45,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.10400000000000001,
      "h": 153.78,
      "l": 0.5961759999999999,
      "mode": "oklch",
    },
  },
  "borderPositiveFaded": {
    "oklch": {
      "c": 0.03,
      "h": 153.78,
      "l": 0.9199999999999999,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.020800000000000003,
      "h": 153.78,
      "l": 0.35,
      "mode": "oklch",
    },
  },
  "borderPrimary": {
    "oklch": {
      "c": 0.24,
      "h": 262.67,
      "l": 0.47000000000000003,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.192,
      "h": 262.67,
      "l": 0.62984,
      "mode": "oklch",
    },
  },
  "borderPrimaryFaded": {
    "oklch": {
      "c": 0.03,
      "h": 262.67,
      "l": 0.9199999999999999,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.038400000000000004,
      "h": 262.67,
      "l": 0.35,
      "mode": "oklch",
    },
  },
  "borderWarning": {
    "oklch": {
      "c": 0.2,
      "h": 80,
      "l": 0.75,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.16000000000000003,
      "h": 80,
      "l": 0.8835999999999999,
      "mode": "oklch",
    },
  },
  "borderWarningFaded": {
    "oklch": {
      "c": 0.05,
      "h": 80,
      "l": 0.9199999999999999,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.03200000000000001,
      "h": 80,
      "l": 0.35,
      "mode": "oklch",
    },
  },
  "brand": {
    "oklch": {
      "c": 0.24,
      "h": 262.67,
      "l": 0.55,
      "mode": "oklch",
    },
    "oklchDark": undefined,
  },
  "foregroundCritical": {
    "oklch": {
      "c": 0.22,
      "h": 26.97,
      "l": 0.5,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.187,
      "h": 26.97,
      "l": 0.75,
      "mode": "oklch",
    },
  },
  "foregroundDisabled": {
    "oklch": {
      "c": 0,
      "h": 262.67,
      "l": 0.84,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0,
      "h": 262.67,
      "l": 0.4,
      "mode": "oklch",
    },
  },
  "foregroundNeutral": {
    "oklch": {
      "c": 0.01,
      "h": 262.67,
      "l": 0.2,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.008,
      "h": 262.67,
      "l": 0.96,
      "mode": "oklch",
    },
  },
  "foregroundNeutralFaded": {
    "oklch": {
      "c": 0.01,
      "h": 262.67,
      "l": 0.45,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.008,
      "h": 262.67,
      "l": 0.8099999999999999,
      "mode": "oklch",
    },
  },
  "foregroundPositive": {
    "oklch": {
      "c": 0.13,
      "h": 153.78,
      "l": 0.5,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.1105,
      "h": 153.78,
      "l": 0.75,
      "mode": "oklch",
    },
  },
  "foregroundPrimary": {
    "oklch": {
      "c": 0.24,
      "h": 262.67,
      "l": 0.5,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.204,
      "h": 262.67,
      "l": 0.75,
      "mode": "oklch",
    },
  },
  "foregroundWarning": {
    "oklch": {
      "c": 0.2,
      "h": 80,
      "l": 0.5,
      "mode": "oklch",
    },
    "oklchDark": {
      "c": 0.17,
      "h": 80,
      "l": 0.75,
      "mode": "oklch",
    },
  },
  "white": {
    "oklch": {
      "c": 0,
      "l": 0.999999993473546,
      "mode": "oklch",
    },
    "oklchDark": undefined,
  },
}
`;
